apps/financial-aid/web-osk/src/routes/status/MainPage/mainPage.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useContext, useMemo } from 'react'
+import {
+  Text,
+  ActionCard,
+  Box,
+  BulletList,
+  Bullet,
+  Button,
+} from '@island.is/island-ui/core'
+
+import {
+  ContentContainer,
+  Footer,
+  StatusLayout,
+} from '@island.is/financial-aid-web/osk/src/components'
+
+import { useRouter } from 'next/router'
+import { UserContext } from '@island.is/financial-aid-web/osk/src/components/UserProvider/UserProvider'
+
+import {
+  ApplicationState,
+  getState,
+  translateMonth,
+} from '@island.is/financial-aid/shared'
+
+import format from 'date-fns/format'
+import { useLogOut } from '@island.is/financial-aid-web/osk/src/utils/useLogOut'
+
+const MainPage = () => {
+  const router = useRouter()
+
+  const { user } = useContext(UserContext)
+  const logOut = useLogOut()
+
+  const currentState = useMemo(() => {
+    if (user?.currentApplication) {
+      return user.currentApplication.state
+    }
+  }, [user])
+
+  const currentApplicationID = useMemo(() => {
+    if (user?.currentApplication) {
+      return user.currentApplication.id
+    }
+  }, [user])
```

Comment:
Lack of knowledge question here 😄 why do we need to use useMemo here?

Reply:
Me and Addi, used this i think is is cleaner coding than using usestate and useeffect